C++17 removed std::auto_ptr

With `CXX=clang++` 16.0.5:
```
In file included from operations.cpp:35:
./stack_cleaner.hpp:75:10: error: no template named 'auto_ptr' in namespace 'std'
    std::auto_ptr< impl > _pimpl;
    ~~~~~^
In file included from stack_cleaner.cpp:32:
./stack_cleaner.hpp:75:10: error: no template named 'auto_ptr' in namespace 'std'
    std::auto_ptr< impl > _pimpl;
    ~~~~~^
1 error generated.
```

<https://en.cppreference.com/w/cpp/memory/auto_ptr>

Temporary fix I did to get it building is adding `-std=c++14` in `CXXFLAGS`, which could be done in lutok itself so you don't have to chase after the latest C++ version and/or end up with `#ifdef` to support multiple C++ versions.

--- a/stack_cleaner.hpp
+++ b/stack_cleaner.hpp
@@ -72,7 +72,7 @@ class stack_cleaner {
     struct impl;
 
     /// Pointer to the shared internal implementation.
-    std::auto_ptr< impl > _pimpl;
+    std::unique_ptr< impl > _pimpl;
 
     /// Disallow copies.
     stack_cleaner(const stack_cleaner&);

